File: VB/WebLayoutSolution.Module/DatabaseUpdate/Updater.vb
Imports System
Imports System.Linq
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.Persistent.BaseImpl
Imports WebLayoutSolution.Module.BusinessObjects

Namespace WebLayoutSolution.Module.DatabaseUpdate
    ' For more typical usage scenarios, be sure to check out http://documentation.devexpress.com/#Xaf/clsDevExpressExpressAppUpdatingModuleUpdatertopic
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            If ObjectSpace.FindObject(Of BusinessObject)(Nothing) Is Nothing Then
                Dim obj = ObjectSpace.CreateObject(Of BusinessObject)()
                obj.FieldA = "aaa"
                obj.FieldB = "bbb"
                obj.FieldC = "ccc"
                obj.FieldD = "ddd"
                obj.FieldE = "eee"
                obj.FieldF = "fff"
                obj.FieldG = "ggg"
                obj.FieldH = "hhh"
                obj.FieldI = "iii"
                obj.FieldJ = "jjj"
                obj.FieldK = "kkk"
                obj.FieldL = "lll"
                obj.FieldM = "mmm"
                obj.FieldN = "nnn"
            End If
        End Sub
        Public Overrides Sub UpdateDatabaseBeforeUpdateSchema()
            MyBase.UpdateDatabaseBeforeUpdateSchema()
            'if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
            '    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
            '}
        End Sub
    End Class
End Namespace

File: CS/WebLayoutSolution.Module/DatabaseUpdate/Updater.cs
using System;
using System.Linq;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Persistent.BaseImpl;
using WebLayoutSolution.Module.BusinessObjects;

namespace WebLayoutSolution.Module.DatabaseUpdate {
    // For more typical usage scenarios, be sure to check out http://documentation.devexpress.com/#Xaf/clsDevExpressExpressAppUpdatingModuleUpdatertopic
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
            base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            if (ObjectSpace.FindObject<BusinessObject>(null) == null) {
                var obj = ObjectSpace.CreateObject<BusinessObject>();
                obj.FieldA = "aaa";
                obj.FieldB = "bbb";
                obj.FieldC = "ccc";
                obj.FieldD = "ddd";
                obj.FieldE = "eee";
                obj.FieldF = "fff";
                obj.FieldG = "ggg";
                obj.FieldH = "hhh";
                obj.FieldI = "iii";
                obj.FieldJ = "jjj";
                obj.FieldK = "kkk";
                obj.FieldL = "lll";
                obj.FieldM = "mmm";
                obj.FieldN = "nnn";
            }
        }
        public override void UpdateDatabaseBeforeUpdateSchema() {
            base.UpdateDatabaseBeforeUpdateSchema();
            //if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
            //    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
            //}
        }
    }
}
